### PR TITLE
Migrate out of Enzyme testing library to facilitate react 18 upgrade

### PR DESCRIPTION
**Pull request by chownces**
### Description

See #2520 

Moving forward, we should probably document down the best practices while onboarding new developers, and point them to the relevant documentation to get started (e.g. React 18, @ testing-library/react, etc.) --> added in #2527 

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### How to test

Check through the snapshots. I've tried adding some assertions to the snapshot tests too to improve the robustness of the test suite.

### Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/commons/sideContent/__tests__/SideContentAutograder.tsx [hunk 1]
```diff
@@ -75,36 +76,29 @@ const resultCardClasses = [
...
-      tree.find(<HTML Selector>) returns two copies of every non-top-level element, as Enzyme
-      returns both React component instances in addition to DOM nodes. This is INTENDED behaviour.
-      For example, the <div> with id of '.testcase-actual' is returned twice since one of them
-      is a React Component, whilst the other is the actual HTML <div> node that Component renders
-    Fix:
-      Append .hostNodes() when desired to only retrieve the HTML DOM nodes from the rendered tree.
-*/
-
-test('Autograder renders placeholders correctly when testcases and results are empty', () => {
+test('Autograder renders placeholders correctly when testcases and results are empty', async () => {
```
Comment (RichDom2185) [inline src/commons/sideContent/__tests__/SideContentAutograder.tsx:79]: Why does this need to be `async`?

File: src/commons/assessment/__tests__/Assessment.tsx [hunk 2]
```diff
@@ -17,79 +20,76 @@ const defaultProps = assertType<AssessmentProps>()({
...
-  // handleSubmitAssessment: (id: number) => {},
-  // isStudent: false,
-  // ...mockRouterProps('/academy/missions', {})
 });
 
-const mockUndefinedAssessment = assertType<AssessmentProps>()({
-  ...defaultProps
-  // assessmentOverviews: undefined
-});
+const getOverridedStore = ({
```
Comment (RichDom2185) [inline src/commons/assessment/__tests__/Assessment.tsx:25]: A minor nit, but the correct word here should be "overridden", not "overrided"

File: src/commons/assessmentWorkspace/__tests__/AssessmentWorkspace.tsx [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-// mock editor for testing update
-jest.mock('../../editor/Editor', () => (props: EditorProps) => (
-  <MockEditor {...props}></MockEditor>
-));
+
+jest.mock('ace-builds', () => ({
+  ...jest.requireActual('ace-builds'),
+  require: jest.fn()
+}));
+const acequireMock = acequire as jest.Mock;
```
Comment (RichDom2185) [inline src/commons/assessmentWorkspace/__tests__/AssessmentWorkspace.tsx:17]: Was this naming choice intentional, or a typo?

**Review by RichDom2185 — APPROVED** [review]
Thanks for all this refactoring work, much appreciated! I only have a few minor nits below, otherwise LGTM 😄

File: src/commons/assessmentWorkspace/__tests__/AssessmentWorkspace.tsx (hunk 3 above)
Comment (chownces) [reply src/commons/assessmentWorkspace/__tests__/AssessmentWorkspace.tsx:17]: intentional as per Editor.tsx, so as to not override the default `require` in JS

**Pull request merged**